shared/src/main/kotlin/com/egm/stellio/shared/model/ApiExceptions.kt

```diff
@@ -2,24 +2,129 @@ package com.egm.stellio.shared.model
+    fun toErrorResponse(): ResponseEntity<ProblemDetail> {
+        logger.info("Returning error ${this.type} (${this.message})")
+        return ResponseEntity.status(status)
+            .contentType(MediaType.APPLICATION_JSON)
+            .body(toProblemDetail())
+    }
+}
+
+data class AlreadyExistsException(override val message: String) : APIException(
+    ErrorType.ALREADY_EXISTS.type,
+    HttpStatus.CONFLICT,
+    message
+)
+
+data class InvalidRequestException(override val message: String) : APIException(
+    ErrorType.INVALID_REQUEST.type,
+    HttpStatus.BAD_REQUEST,
+    message
+)
+
+data class BadRequestDataException(override val message: String) : APIException(
+    ErrorType.BAD_REQUEST_DATA.type,
+    HttpStatus.BAD_REQUEST,
+    message
+)
+data class OperationNotSupportedException(override val message: String) : APIException(
+    ErrorType.OPERATION_NOT_SUPPORTED.type,
+    HttpStatus.BAD_REQUEST,
+    message
+)
+data class ResourceNotFoundException(override val message: String) : APIException(
+    ErrorType.RESOURCE_NOT_FOUND.type,
+    HttpStatus.NOT_FOUND,
+    message
+)
+data class InternalErrorException(override val message: String) : APIException(
+    ErrorType.INTERNAL_ERROR.type,
+    HttpStatus.INTERNAL_SERVER_ERROR,
+    message
+)
+
+data class TooManyResultsException(override val message: String) : APIException(
+    ErrorType.TOO_MANY_RESULTS.type,
+    HttpStatus.FORBIDDEN,
+    message
+)
+data class AccessDeniedException(override val message: String) : APIException(
+    ErrorType.ACCESS_DENIED.type,
+    HttpStatus.FORBIDDEN,
+    message
+)
+data class NotImplementedException(override val message: String) : APIException(
+    ErrorType.NOT_IMPLEMENTED.type,
+    HttpStatus.NOT_IMPLEMENTED,
+    message
+)
+data class LdContextNotAvailableException(override val message: String) : APIException(
+    ErrorType.LD_CONTEXT_NOT_AVAILABLE.type,
+    HttpStatus.SERVICE_UNAVAILABLE,
+    message
+)
+data class NonexistentTenantException(override val message: String) : APIException(
+    ErrorType.NONEXISTENT_TENANT.type,
+    HttpStatus.NOT_FOUND,
+    message
+)
+data class TooComplexQueryException(override val message: String) : APIException(
+    ErrorType.TOO_COMPLEX_QUERY.type,
+    HttpStatus.FORBIDDEN,
+    message
+)
+data class NotAcceptableException(override val message: String) : APIException(
+    ErrorType.NOT_ACCEPTABLE.type,
+    HttpStatus.NOT_ACCEPTABLE,
+    message
+)
+
+data class JsonParseApiException(override val message: String) : APIException(
+    ErrorType.INVALID_REQUEST.type,
+    HttpStatus.BAD_REQUEST,
+    message
+)
+
+data class UnsupportedMediaTypeStatusApiException(override val message: String) : APIException(
+    ErrorType.UNSUPPORTED_MEDIA_TYPE.type,
+    HttpStatus.UNSUPPORTED_MEDIA_TYPE,
+    message
+)
+
+data class JsonLdErrorApiResponse(override val message: String, override val detail: String) : APIException(
+    ErrorType.BAD_REQUEST_DATA.type, // todo check
```

Comment:
yes: "If the data included by the JSON-LD document is not syntactically correct, according to the @context or the API data type definitions, then an error of type BadRequestData shall be raised"